src/index.ts

```diff
@@ -144,6 +145,10 @@ export class Cinnamon {
         headers?: Headers;
         token?: string;
     }): Promise<APIResult<T, U>> {
+        if (!get(variables, 'input.refreshToken')) {
```

Comment:
You don't need this condition right? could always await `this.refreshToken` and if is `null` it will continue the execution.

Reply:
we need it, or the refreshToken call waits for itself.

Reply:
It doesn't, the function `refreshLogin` calls `api` before it returns, in fact the lock won't stop concurrent calls from getting an error (`this.refreshTokenError` is `null` for all of them), but they end up resolving well because they will see the peding promise inside `this.refreshTokenError` down bellow because the `await fetch` call defers the code after it.

Reply:
this works when you call them in the same tick, but not if the next call comes while the request is in progress.

this is hard to test manually, thats why i wrote the unit test. The test fails if you remove this.

Reply:
but yea we can remove the if check, it should not hang from itself.